frontend/src/metabase/visualizations/visualizations/SmartScalar/PreviousValueComparison/PreviousValueComparison.tsx

```diff
@@ -0,0 +1,144 @@
+import innerText from "react-innertext";
+
+import DashboardS from "metabase/css/dashboard.module.css";
+import { formatValue } from "metabase/lib/formatting/value";
+import { measureTextWidth } from "metabase/lib/measure-text";
+import { Badge, Flex, Group, Icon, Stack, Tooltip } from "metabase/ui";
+import type { ColumnSettings } from "metabase/visualizations/types";
+
+import { CHANGE_TYPE_OPTIONS, type ComparisonResult } from "../compute";
+import {
+  ELLIPSIS_BADGE_WIDTH,
+  ICON_MARGIN_RIGHT,
+  ICON_SIZE,
+  SPACING,
+  TEXT_SPACING,
+} from "../constants";
+import { formatChangeAutoPrecision, getChangeWidth } from "../utils";
+
+import { DetailCandidate } from "./DetailCandidate";
+import { PreviousValueComparisonTooltip } from "./PreviousValueComparisonTooltip";
+import { VariationDetails } from "./VariationDetails";
+import { VariationPercent } from "./VariationPercent";
+
+interface PreviousValueComparisonProps {
+  comparison: ComparisonResult;
+  fontFamily: string;
+  formatOptions: ColumnSettings;
+  tooltipComparisons: ComparisonResult[];
+  width: number;
+}
+
+export function PreviousValueComparison({
+  comparison,
+  fontFamily,
+  formatOptions,
+  tooltipComparisons,
+  width,
+}: PreviousValueComparisonProps) {
+  const fontSize = "0.875rem";
+  const { changeType, percentChange, comparisonValue, display } = comparison;
+  const showsOtherValuesInTooltip = tooltipComparisons.length > 1;
+
+  const fittedChangeDisplay =
```

Comment:
`fittedChangeDisplay` and `availableComparisonWidth` require some heavy computations, so I suggest to wrap the calculations with useMemo

Reply:
Actually, most of the variables declared here would benefit from `useMemo`. Specifically, those using `measureTextWidth` and `formatValue`.